File: 10TypeScript/Ex1_1.ts
class Ex1_1 {
    main() {
        let a: number = 1;
        let b: number = 2;
        let c: number = 3;
        let d: number = 4;

        let t: number = a;

        a = b;
        b = c;
        c = d;
        d = t;

        return a + ", " + b + ", " + c + ", " + d;
    }
}

let app = new Ex1_1();

console.log(app.main());

File: 08Ruby/ex1_1.rb
class App
  def main
    a = 1
    b = 2
    c = 3
    d = 4

    t = a

    a = b
    b = c
    c = d
    d = t

    puts "#{a}, #{b}, #{c}, #{d}"
  end
end

if __FILE__ == $0
  app = App.new
  app.main
end